BibBuddy/app/src/main/java/de/bibbuddy/AsDefaultAppFragment.java

```diff
@@ -0,0 +1,500 @@
+    libraryRecyclerView.setAdapter(adapter);
+    libraryRecyclerView.setListener(this);
+
+    updateEmptyView(libraryList);
+  }
+
+  private void setupAddShelfBtn() {
+    FloatingActionButton addShelfBtn = view.findViewById(R.id.add_btn);
+    createAddShelfListener(addShelfBtn);
+  }
+
+  private void createAddShelfListener(FloatingActionButton addShelfBtn) {
+    addShelfBtn.setOnClickListener(v -> handleAddShelf());
+  }
+
+  private Bundle createAddShelfBundle() {
+    Bundle bundle = new Bundle();
+
+    Long currentShelfId = libraryModel.getShelfId();
+    if (currentShelfId == null) {
+      bundle.putLong(LibraryKeys.SHELF_ID, 0L);
+    } else {
+      bundle.putLong(LibraryKeys.SHELF_ID, currentShelfId);
+    }
+
+    bundle.putStringArray(LibraryKeys.SHELF_NAMES, getAllShelfNames());
+
+    return bundle;
+  }
+
+  private String[] getAllShelfNames() {
+    List<ShelfItem> currentLibraryList = libraryModel.getCurrentLibraryList();
+    String[] shelfNames = new String[currentLibraryList.size()];
+
+    for (int i = 0; i < currentLibraryList.size(); i++) {
+      shelfNames[i] = currentLibraryList.get(i).getName();
+    }
+
+    return shelfNames;
+  }
+
+  private void handleAddShelf() {
+    LibraryFormFragment libraryFormFragment =
+        new LibraryFormFragment(new LibraryFormFragment.ChangeShelfListener() {
+          @Override
+          public void onShelfAdded(String name, Long shelfId) {
+            libraryModel.addShelf(name, libraryModel.getShelfId());
+            updateLibraryListView(libraryModel.getCurrentLibraryList());
+            Toast.makeText(context, getString(R.string.shelf_added), Toast.LENGTH_SHORT).show();
+          }
+        });
+
+    libraryFormFragment.setArguments(createAddShelfBundle());
+
+    showFragment(libraryFormFragment, LibraryKeys.FRAGMENT_LIBRARY_FORM);
+  }
+
+  private void updateEmptyView(List<ShelfItem> libraryList) {
+    TextView emptyView = view.findViewById(R.id.list_view_library_empty);
+
+    if (libraryList.isEmpty()) {
+      emptyView.setVisibility(View.VISIBLE);
+    } else {
+      emptyView.setVisibility(View.GONE);
+    }
+  }
+
+  private void updateLibraryListView(List<ShelfItem> libraryList) {
+    libraryList = libraryModel.getSortedLibraryList(sortCriteria, libraryList);
+    adapter.notifyDataSetChanged();
+    updateEmptyView(libraryList);
+  }
+
+  private void updateBookListView(LibraryItem libraryItem) {
+    addLibraryFragmentToBackStack();
+    BookFragment fragment = new BookFragment();
+    fragment.setArguments(createShelfBundle(libraryItem));
+    showFragment(fragment);
+  }
+
+  private void addLibraryFragmentToBackStack() {
+    Fragment libraryFragment = new LibraryFragment();
+    FragmentTransaction ft = requireActivity()
+        .getSupportFragmentManager().beginTransaction();
+    ft.replace(R.id.fragment_container_view, libraryFragment);
+    ft.addToBackStack(null);
+    ft.commit();
```

Comment:
Else there will be errors.
```suggestion
    Fragment libraryFragment = new LibraryFragment();
    showFragment(libraryFragment);
```